compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -1805,8 +1805,11 @@ impl<'tcx> LayoutCx<'tcx, TyCtxt<'tcx>> {
 
         match layout.variants {
             Variants::Single { index } => {
-                debug!("print-type-size `{:#?}` variant {}", layout, adt_def.variants[index].ident);
-                if !adt_def.variants.is_empty() {
+                if !adt_def.variants.is_empty() && layout.fields != FieldsShape::Primitive {
```

Comment:
Why is variants non-empty here? Primitives aren't enums.

Reply:
FieldsShape describes how fields are located in memory, with Primitive used for things that have no fields, including the never type.

Variants that are uninhabited and have ZST fields only are consider to be absent. Layout computation omits absents variants entirely. If all variants of an enum are absent, the layout of a never type is used (Variants::Single, FieldsShape::Primitive).